lib/plugins/azurepush.js

```diff
@@ -0,0 +1,323 @@
+        if (docs.length > 0) {
+          ctx.store.collection(env.azurepush_collection).update({_id: entry._id}, {$set: {settings: entry.settings}}, function updateAzurepush(err, doc) {
+            // *****REMOVE*****: Beta logging
+            if (err) {
+              logTrelloIssue(env, '0x05-Failed to update existing registration in mongodb');
+            }
+            // END: *****REMOVE*****
+
+            // TODO: Handle error
+            console.log(err, doc);
+          });
+        } else {
+          ctx.store.collection(env.azurepush_collection).insert(entry, function storeAzurepush(err, doc) {
+            // *****REMOVE*****: Beta logging
+            if (err) {
+              logTrelloIssue(env, '0x06-Failed to add new registration in mongodb');
+            }
+            // END: *****REMOVE*****
+            
+            // TODO: Handle error
+            console.log(err, doc);
+          });
+        }
+      });
+
+      // Verify/create index
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.announcement', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x07-Failed to verify index on anncouncement in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.alert', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x08-Failed to verify index on alert in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.info', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x09-Failed to verify index on info in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      
+      // More info on request body: https://msdn.microsoft.com/en-us/library/mt621153.aspx
+      var installation = {
+        installationId: request.installationId
+        , pushChannel: request.deviceToken
+        , templates: {
+          nightscout: {
+            tags: tags
+          }
+        }
+      };
+
+      // 'platform': https://developer.xamarin.com/guides/xamarin-forms/platform-features/device/#Device.OS
+      switch (request.platform) {
+        case 'iOS':
+          installation.platform = 'apns';
+          installation.templates.nightscout.body = "{\"aps\":{\"alert\":\"$(message)\",\"sound\":\"$(sound)\",\"category\":\"$(category)\"},\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"title\":\"$(title)\"}";
```

Comment:
I think I can change the outer most quotes to single quotes, but can't remove the escaping or use stringify on a JSON object.  

The string for `body` is actually used in Azure to translate an actual JSON object sent in a notification (from Nightscout, which isn't stringify'd or build as a string) to the JSON payload sent to the messaging platform.  In otherwords, I'm not building a JSON object for `body` so much as I'm building a map for how Azure will translate my actual payload into a platform specific notification.

I'll at least update the outer most quotes to single quotes to keep with coding style.